plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTablePqlExtractor.java

```diff
@@ -92,7 +95,11 @@ public static String extractPql(DynamicTable table, TupleDomain<ColumnHandle> tu
             }
             builder.append(table.limit().getAsLong());
         }
-        return builder.toString();
+        String pql = builder.toString();
+        if (log.isDebugEnabled()) {
+            log.debug("Generated PQL: %s", pql);
```

Comment:
The logging place seems wrong. We should log the generated PQL just before the execution.